### PR TITLE
Fix for Arrow Scans

### PR DESCRIPTION
**Pull request by pdet**
In this PR:
* Add Arrow Tests to CI.
* Aquire locks when scanning arrow objects.
* Partition python's test files to folders.
* Also build FTS when building python since this is now a dependency to it.

### CODE REVIEW

File: src/include/duckdb/common/helper.hpp [hunk 1]
```diff
@@ -18,14 +18,14 @@
 namespace duckdb {
 #if !defined(_MSC_VER) && (__cplusplus < 201402L)
 template <typename T, typename... Args>
-unique_ptr<T> make_unique(Args &&...args) {
+unique_ptr<T> make_unique(Args &&... args) {
```
Comment (hannesmuehleisen) [inline src/include/duckdb/common/helper.hpp:21]: Please don't check in this change, its a flip-flop between versions of clang-format

File: tools/pythonpkg/src/pyconnection.cpp [hunk 2]
```diff
@@ -245,6 +245,7 @@ struct PythonTableArrowArrayStream {
 
 	static int MyStreamGetSchema(struct ArrowArrayStream *stream, struct ArrowSchema *out) {
 		D_ASSERT(stream->private_data);
+		py::gil_scoped_acquire acquire;
```
Comment (hannesmuehleisen) [inline tools/pythonpkg/src/pyconnection.cpp:248]: I realise that this change stops it crashing, but we should double-check why that is the case.

File: src/include/duckdb/common/helper.hpp (hunk 1 above)
Comment (hannesmuehleisen) [reply src/include/duckdb/common/helper.hpp:21]: There is another one or two like it

File: tools/pythonpkg/src/pyconnection.cpp (hunk 2 above)
Comment (hannesmuehleisen) [reply tools/pythonpkg/src/pyconnection.cpp:248]: by double-check I mean find out

File: src/include/duckdb/common/helper.hpp (hunk 1 above)
Comment (pdet) [reply src/include/duckdb/common/helper.hpp:21]: I agree, of course, I can look at it, but of course, this won't solve the underlying problem. 
Maybe enforcing a clang-format version is an option?

Comment (hannesmuehleisen) [reply src/include/duckdb/common/helper.hpp:21]: I just don't add those particular changes to my commit

Comment (Mytherin) [reply src/include/duckdb/common/helper.hpp:21]: Let's just exclude these three files from the make format-fix?

**Comment by hannesmuehleisen** [comment]
Looks like the `cmake` in the cibuildwheel is too old to build pyarrow - hooray ^^ `CMake 3.2 or higher is required.  You are running version 2.8.12.2`

**Comment by pdet** [comment]
So, I've tried upgrading cmake, but this caused some linking issues for the cp36-manylinux_i686 wheel, in the end, I changed it to try to install pyarrow, if it fails skip the tests, so the CI runs the arrow tests on mac on most Linux wheels

**Pull request merged**